feat: get all required dependencies in setup.py from issue #105

HanslettTheDev opened the pull request:
## Summary

This pull request solves this [Issue](https://github.com/PyAr/pyafipws/issues/105). The setup.py file has some other dependencies which setup.py has to be aware of so as to build the project. This PR solves that issue

## Checklist

- [x] Classes, Variables, function and methods logic  ok
- [x] Comments written explaining what the code does
- [x] All python code is PEP8 compliant (run black .)
- [x] No lint issues (run flake8)


## Manual test evidence

(attach command-line examples, execution output & logs, etc.)
![image](https://user-images.githubusercontent.com/64011386/231130123-ae8fd88c-5fda-49dc-9e3d-505b70dafa20.png)

HanslettTheDev committed:
feat: get all required dependencies in setup.py from issue #105

Signed-off-by: HanslettTheDev <HanslettTheDev@gmail.com>

HanslettTheDev committed:
feat: get all required dependencies in setup.py from issue #105

Signed-off-by: HanslettTheDev <HanslettTheDev@gmail.com>

reingart commented on get_dep.py:
```diff
@@ -0,0 +1,8 @@
+
+def get_dependecies() -> list:
+    # The subprocess.Popen is used for backward compatibility with python 2.7
+    freeze = subprocess.Popen("pip freeze", shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE).communicate()
```
`pip freeze` probably will not work if you download this project as a package from pypi.
You can test it running `pip install .` in an empty virtual environment.
Instead you could read the requirements.txt or just passing a list of explicit dependencies as shown in the ticket #105

reingart reviewed (changes requested):
Thanks for the PR!
I didn't see `pip freeze` used in this context, did you have any specific reason to use it?
Nice first approach, please see the review comments in each file and try to address them

HanslettTheDev committed:
fix: improved the get_dependencies function to grab all dependency modules

Signed-off-by: HanslettTheDev <HanslettTheDev@gmail.com>

HanslettTheDev committed:
fix: improved the get_dependencies function to grab all dependency modules; commit signed

Signed-off-by: HanslettTheDev <HanslettTheDev@gmail.com>

HanslettTheDev committed:
fix: fixed the issue with pip not detecting links

Signed-off-by: HanslettTheDev <HanslettTheDev@gmail.com>

HanslettTheDev committed:
fix: removed the python version number in the requirements.txt file
fix: fixed the error when replacing the links with the module name
The overall errors fixed was to prevent the setup.py build from failing

Signed-off-by: HanslettTheDev <HanslettTheDev@gmail.com>

HanslettTheDev commented:
Sorry, I had to rebase my branch. 
I am not used to the signed commits. So I had to rebase then push the signed commits
